portal-impl/src/com/liferay/portal/service/impl/UserLocalServiceImpl.java

```diff
@@ -6334,8 +6334,9 @@ protected void notifyUser(User user, ServiceContext serviceContext)
 
 			passwordResetURL = StringBundler.concat(
 				serviceContext.getPortalURL(), serviceContext.getPathMain(),
-				updatePasswordURL, "ticketId=", ticket.getTicketId(),
-				"&ticketKey=", ticket.getKey());
+				"&languageId=", user.getLanguageId(), updatePasswordURL,
```

Comment:
It's not correct also in this version. Could you please have a look? (and please verify that the url produced is valid and working)

Reply:
@smottal
I carried out the test by placing the parameter at all points, and the only one that is functional is as in the first version, languageId in the last parameter

Reply:
Don't work too

```
			passwordResetURL = StringBundler.concat(
				serviceContext.getPortalURL(), serviceContext.getPathMain(),
				updatePasswordURL, "&languageId=", user.getLanguageId(),
				"&ticketId=", ticket.getTicketId(), "&ticketKey=",
				ticket.getKey());
```

Reply:
@henriquelimadotcom the URL should have this format
protocol://basePath/url **?** param1=value1 **&** param2=value2...
in this case
serviceContext.getPortalURL() + serviceContext.getPathMain() + updatePasswordURL + parameters

if you see the **?** is already at the end of updatePasswordURL (for this reason the first original parameter ticketId was without any char before) so we don't need to write it.

In order to have this URL generated correctly I think you should do something like this
```
passwordResetURL = StringBundler.concat(
				serviceContext.getPortalURL(), serviceContext.getPathMain(),
				updatePasswordURL, "languageId=", user.getLanguageId(),
				"&ticketId=", ticket.getTicketId(), "&ticketKey=",
				ticket.getKey());
```
please verify it and resend. Thx